From a47e583096c9d9187aaca0b8b33694baaf72e63f Mon Sep 17 00:00:00 2001
From: Jun Li <r65092@freescale.com>
Date: Fri, 12 Jun 2009 15:30:30 +0800
Subject: [PATCH] ENGR00113284 use late_initcall to fix usb gadget drvers init issue.

When usb gadet is built as module, usb gadget functions
can not work. This patch replace device_initcall_sync
with late_initcall to fix usb gadget drivers init issue.

Signed-off-by: Li Jun <r65092@freescale.com>
---
 drivers/usb/gadget/arcotg_udc.c   |    2 +-
 drivers/usb/gadget/ether.c        |    2 +-
 drivers/usb/gadget/file_storage.c |    2 +-
 drivers/usb/gadget/serial.c       |    2 +-
 drivers/usb/host/ehci-hcd.c       |    2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/usb/gadget/arcotg_udc.c b/drivers/usb/gadget/arcotg_udc.c
index 2d3f89b..af7b940 100644
--- a/drivers/usb/gadget/arcotg_udc.c
+++ b/drivers/usb/gadget/arcotg_udc.c
@@ -2950,7 +2950,7 @@ static int __init udc_init(void)
 	return platform_driver_register(&udc_driver);
 }
 
-device_initcall_sync(udc_init);
+late_initcall(udc_init);
 
 static void __exit udc_exit(void)
 {
diff --git a/drivers/usb/gadget/ether.c b/drivers/usb/gadget/ether.c
index 37252d0..8e34d9d 100644
--- a/drivers/usb/gadget/ether.c
+++ b/drivers/usb/gadget/ether.c
@@ -385,7 +385,7 @@ static int __init init(void)
 {
 	return usb_composite_register(&eth_driver);
 }
-module_init(init);
+late_initcall(init);
 
 static void __exit cleanup(void)
 {
diff --git a/drivers/usb/gadget/file_storage.c b/drivers/usb/gadget/file_storage.c
index 2529b1f..58e278e 100644
--- a/drivers/usb/gadget/file_storage.c
+++ b/drivers/usb/gadget/file_storage.c
@@ -4148,7 +4148,7 @@ static int __init fsg_init(void)
 		kref_put(&fsg->ref, fsg_release);
 	return rc;
 }
-module_init(fsg_init);
+late_initcall(fsg_init);
 
 
 static void __exit fsg_cleanup(void)
diff --git a/drivers/usb/gadget/serial.c b/drivers/usb/gadget/serial.c
index 37879af..e7bd43d 100644
--- a/drivers/usb/gadget/serial.c
+++ b/drivers/usb/gadget/serial.c
@@ -273,7 +273,7 @@ static int __init init(void)
 
 	return usb_composite_register(&gserial_driver);
 }
-module_init(init);
+late_initcall(init);
 
 static void __exit cleanup(void)
 {
diff --git a/drivers/usb/host/ehci-hcd.c b/drivers/usb/host/ehci-hcd.c
index 6a35d6c..058219e 100644
--- a/drivers/usb/host/ehci-hcd.c
+++ b/drivers/usb/host/ehci-hcd.c
@@ -1125,7 +1125,7 @@ err_debug:
 	clear_bit(USB_EHCI_LOADED, &usb_hcds_loaded);
 	return retval;
 }
-device_initcall_sync(ehci_hcd_init);
+late_initcall(ehci_hcd_init);
 
 static void __exit ehci_hcd_cleanup(void)
 {
-- 
1.5.4.4

